src/WorldMap.h

```diff
@@ -1,27 +1,96 @@
 #pragma once
 
-#include "TileLayer.h"
-#include "TileSheetDetails.h"
-#include <vector>
+#include <unordered_map>
 #include <SFML\Graphics.hpp>
 #include <string>
 
+class TiXmlElement;
+struct LevelDetails;
 class WorldMap
 {
+	struct TileSheetDetails
+	{
+		TileSheetDetails(std::string&& name, int tileSize, int rows, int columns, double margin, double spacing);
+
+		const std::string m_name;
+		const int m_tileSize;
+		const int m_rows;
+		const int m_columns;
+		const int m_margin;
+		const int m_spacing;
+	};
+	
+	class TileSheet
+	{
+	public:
+		TileSheet(const TileSheetDetails& tileSheetDetails, const std::string& tileSheetPath);
+		~TileSheet();
+		TileSheet(const TileSheet& orig);
+
+		const std::string& getFilePath() const;
+		const TileSheetDetails& getDetails() const;
+		const sf::Texture& getTexture() const;
+
+		sf::IntRect getTileLocationByID(int ID) const;
+		sf::IntRect getTileLocationByPosition(const sf::IntRect& rect) const;
+
+	private:
+		const TileSheetDetails m_tileSheetDetails;
+		const std::string& m_filePath;
```

Comment:
Doesn't seem worth it to make this a reference.  We're not creating heaps of tilesheets, so a string copy when we create one isn't a bit hit.  On the other hand, dangling reference bugs are very annoying to deal with.